orangecrab/src/Ila.hs

```diff
@@ -110,15 +178,26 @@ ila ::
   , 1 <= BitSize a `DivRU` 8
   , 1 <= size
   ) =>
-  -- | Maximum sample count
-  SNat size ->
+  -- | The configuration of the ILA
+  IlaConfig size (Signal dom a) ->
   -- | Trigger predicate
   (a -> Bool) ->
-  -- | Signal to probe
-  Signal dom a ->
-  -- | Circuit outputting the content of the buffer whenever the predicate has been triggered
-  -- The input signal is to clear the trigger
+  -- | The ILA circuit, the input needs to be connected to some sort of byte input and the output
+  -- is a PacketStream containing bytes to be routed to the PC.
+  --
+  -- TODO: example? Not added one yet due to possible change in API
   Circuit
-    (CSignal dom Bool)
+    (CSignal dom (Maybe (BitVector 8)))
```

Comment:
If you just want to represent the input as a sequence of bytes I think it's more idiomatic to prefer `Df dom (BitVector 8)` over `CSignal dom (Maybe (BitVector 8))` unless there is a fundamental reason why you can't supply backpressure